README.md

```diff
@@ -226,6 +229,51 @@ The only required key/value in the signing_info dictionary is 'identity'.
 See the **SIGNED PACKAGES** section of the man page for `pkgbuild` or the **SIGNED PRODUCT ARCHIVES** section of the man page for `productbuild` for more information on the signing options.
 
 
+### Package notarization
+
+You may notarize **SIGNED PACKAGES** as part of the build process by adding a notarization dictionary to the build\_info.plist:
+
+```plist
+    <key>notarization</key>
+    <dict>
+        <key>username</key>
+        <string>john.appleseed@apple.com</string>
+        <key>password</key>
+        <string>@keychain:AC_PASSWORD</string>
+        <key>stapler_timeout</key>
+        <integer>120</integer>
+    </dict>
+```
+
+or, in JSON format in a build-info.json file:
+
+```json
+    "notarization": {
+        "username": "john.appleseed@apple.com",
+        "password": "@keychain:AC_PASSWORD",
+        "stapler_timeout":  120
+    }
+```
+
+The only required keys/values in the notarization dictionary are `username` and `password`:
+
+- `username` is the login email adress of your developer Apple ID.
+- `password` is the 2FA app specific password. For information about the password and saving
+it to the login keychain see the web page [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow).
+
+
+Notarization process is described on Apple web page [Customizing the Notarization Workflow](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow).
+
+`munki-pkg` basically runs two commands:
+
+```shell
+xcrun altool --notarize-app --primary-bundle-id "com.github.munki.pkg.munki-kickstart" --username "john.appleseed@apple.com" --password "@keychain:AC_PASSWORD" --file munki_kickstart.pkg
+xcrun stapler staple munki_kickstart.pkg
+```
```

Comment:
We need to make sure document/point out all the dependencies/requirements that notarization support adds. Like needing to have Xcode  -- or will the Command Line Tools suffice? -- and of course unfettered network access to Apple, etc. I don't want to have to provide support for that sort of thing, but if it's not clearly documented (and probably even if it is) there will be FAQs...

Reply:
Also need to warn people that the first time they access a given keychain item with this method they'll need to be logged in at the GUI to authorize a keychain access prompt.

Reply:
When I tried this workflow, I got this error:

```
2019-06-06 14:43:32.599 altool[66084:6324246] *** Error: Your Apple ID account is attached to other iTunes providers. You will need to specify which provider you intend to submit content to by using the -itc_provider command. Please contact us if you have questions or need help. (1627)
```

I'm probably not the only person whose AppleID is associated with more than one organization, so we'll need to provide a method to provide `-itc_provider` info.

Reply:
Error message is incorrect, the flag needed is `--asc-provider`. The value you need to provide is not immediately obvious and I'll post about that in a bit.

Reply:
I got a list of the proper identifiers for the providers like so:
```
/Applications/Xcode.app/Contents/Applications/Application\ Loader.app/Contents/itms/bin/iTMSTransporter -m provider -u 'gregneagle@mac.com' -p 'secret-app-specific-password' -account_type itunes_connect -v off
```
I was unsuccessful in using a "@keychain:AC_PASSWORD" reference for the required password; instead I had to use the actual app-specific-password.  This is probably not that important since this should be a one-time thing and not something that needs to be automated.

The above command gave me the following output:

```
Provider listing:
   - Long Name -                               - Short Name -
1  Greg Neagle|2XXXXXX5                        GregNeagle2XXXXXX5
2  Walt Disney Animation Studios - CXXXXXXXXT  CXXXXXXXXT
```

I was then able to run `xcrun altool --notarize-app` with `--asc-provider "GregNeagle2XXXXXX5"`

This failed later due to "The executable does not have the hardened runtime enabled." and that's going to be another source of support questions...

Reply:
> This failed later due to "The executable does not have the hardened runtime enabled." and that's going to be another source of support questions...

Did you tried to notarize package with binaries/app bundles/etc. inside? That might explain the`hardened runtime` message. Will look into it.

>  We need to make sure document/point out all the dependencies/requirement

I will add documentation about the dependencies.

> Also need to warn people that the first time they access a given keychain item with this method they'll need to be logged in at the GUI to authorize a keychain access prompt.
Perhaps paragraph about keychain password should be added?

> --asc-provider

Ok. This needs to be added to the `notarization_info` dicitionary.

Reply:
Added support for --asc-provider. See de17997109a31e8ea22836aeccfc38d21d6cdd73

Reply:
Added note about Xcode 10+ requirment. See 806aefb19e395e4a9e4fb22d338bb9ddd8eeb8a8

Enhanced error handling when `xcrun` is unable to run `altool. See 63111cd718eafc4d55c138d31c832939f4769f90.

Reply:
> "This failed later due to "The executable does not have the hardened runtime enabled." and that's going to be another source of support questions..."
Did you tried to notarize package with binaries/app bundles/etc. inside? That might explain thehardened runtime message. Will look into it.

My point was simply that people are going to try to notarize packages that have payloads that contain items that will fail notarization. This is going to be a source of support questions. People won't realize that all code in the payload (including but not limited to app bundles, frameworks, kernel extensions) needs to be properly signed with the hardened runtime restrictions in order to be notarized. IOW when you attempt to notarize a _package_ Apple wants to notarize app bundles, frameworks, kernel extensions (and maybe more) in the package payload as well.

Reply:
Updated the README. This should be a bit better. See fixup dc0adc8

Reply:
And fixup of that fixup (sorry about that) c5917ff adds even more information.